pubsub/publish.go

```diff
@@ -2,30 +2,51 @@ package pubsub
 
 import (
 	"context"
+	"log"
+	"os"
 
-	gcloud "cloud.google.com/go/pubsub"
+	gcp "cloud.google.com/go/pubsub"
 )
 
-const (
-	projectID = "athenian-1"
-	topicID   = "metadata"
+var (
+	// projectID is set from the GCP_PROJECT environment variable, which is
+	// automatically set by the Cloud Functions runtime.
+	projectID = os.Getenv("GCP_PROJECT")
 )
 
-func Publish(ctx context.Context, payload []byte) error {
-	cli, err := gcloud.NewClient(ctx, projectID)
-	if err != nil {
-		return err
-	}
+// Publisher is Google Pub/Sub publisher.
+type Publisher struct {
+	topic *gcp.Topic
+}
 
-	topic := cli.Topic(topicID)
-	msg := &gcloud.Message{
-		Data: payload,
+// NewPublisher creates a new instance of Pub/Sub publisher.
+func NewPublisher(topicID string) (*Publisher, error) {
+	ctx := context.Background()
+
+	client, err := gcp.NewClient(ctx, projectID)
+	if err != nil {
+		return nil, err
 	}
 
-	_, err = topic.Publish(ctx, msg).Get(ctx)
+	// Create the topic if it doesn't exist.
+	topic := client.Topic(topicID)
+	exists, err := topic.Exists(ctx)
 	if err != nil {
-		return err
+		return nil, err
 	}
+	if !exists {
+		if _, err = client.CreateTopic(ctx, topicID); err != nil {
+			return nil, err
+		}
+	}
+
+	return &Publisher{topic: topic}, nil
+}
+
+// Publish publishes data to the Pub/Sub topic synchronously.
```

Comment:
```suggestion
// Publish posts data to the Pub/Sub topic synchronously.
```
(avoiding tautology in the docs)